test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,597 @@
+    dispatcher_params = list(dispatcher_signature.parameters.values())[1:]
+
+    datapoint_method = getattr(datapoints._datapoint.Datapoint, dispatcher.__name__)
+    datapoint_signature = inspect.signature(datapoint_method)
+    datapoint_params = list(datapoint_signature.parameters.values())[1:]
+
+    # Because we use `from __future__ import annotations` inside the module where `datapoints._datapoint` is
+    # defined, the annotations are stored as strings. This makes them concrete again, so they can be compared to the
+    # natively concrete dispatcher annotations.
+    datapoint_annotations = get_type_hints(datapoint_method)
+    for param in datapoint_params:
+        param._annotation = datapoint_annotations[param.name]
+
+    assert dispatcher_params == datapoint_params
+
+
+def check_dispatcher_signatures_match(dispatcher, *, kernel, input_type):
+    _check_dispatcher_kernel_signature_match(dispatcher, kernel=kernel, input_type=input_type)
+    _check_dispatcher_datapoint_signature_match(dispatcher)
+
+
+def _check_transform_v1_compatibility(transform):
+    if not hasattr(transform, "_v1_transform_cls"):
+        return
+
+    if hasattr(transform._v1_transform_cls, "get_params"):
+        assert type(transform).get_params is transform._v1_transform_cls.get_params
+
+    _script(transform)
+
+
+def check_transform(transform_cls, input, *args, **kwargs):
+    transform = transform_cls(*args, **kwargs)
+
+    output = transform(input)
+    assert isinstance(output, type(input))
+
+    if isinstance(input, datapoints.BoundingBox):
+        assert output.format == input.format
+
+    _check_transform_v1_compatibility(transform)
+
+
+# We cannot use `list(transforms.InterpolationMode)` here, since it includes some PIL-only ones as well
+INTERPOLATION_MODES = [
+    transforms.InterpolationMode.NEAREST,
+    transforms.InterpolationMode.NEAREST_EXACT,
+    transforms.InterpolationMode.BILINEAR,
+    transforms.InterpolationMode.BICUBIC,
+]
+
+
+@contextlib.contextmanager
+def assert_warns_antialias_default_value():
+    with pytest.warns(UserWarning, match="The default value of the antialias parameter of all the resizing transforms"):
+        yield
+
+
+class TestResize:
+    SPATIAL_SIZE = (17, 11)
+    SIZES = [17, [17], (17,), [12, 13], (12, 13)]
+
+    def _make_max_size_kwarg(self, *, use_max_size, size):
+        if use_max_size:
+            if not (isinstance(size, int) or len(size) == 1):
+                # This would result in an `ValueError`
+                return None
+
+            max_size = (size if isinstance(size, int) else size[0]) + 1
+        else:
+            max_size = None
+
+        return dict(max_size=max_size)
+
+    def _make_input(self, input_type, *, dtype=None, device="cpu"):
+        if input_type in {torch.Tensor, PIL.Image.Image, datapoints.Image}:
+            input = make_image(size=self.SPATIAL_SIZE, dtype=dtype or torch.uint8, device=device)
```

Comment:
Best not to use `make_image` as it is extremely complex. Is that a simple call to `torch.randint()`?

Reply:
It is indeed complex right now, but it currently also has to support the v2 machinery that we are trying to replace. When that is done, the stuff left will be 

https://github.com/pytorch/vision/blob/3d70e4bb9ee4197a7271ec88f567a395472697e3/test/common_utils.py#L505-L512

which is not too hard.

> Is that a simple call to `torch.randint()`?

Only for integer images. But since we also want to test floating point images, we either need to switch to `torch.rand` for them, or use `make_tensor` as above.

Reply:
Probably best to extract that now then, so we're sure moving forward that we're never relying on some complex machinery from `make_image`

Reply:
See https://github.com/pytorch/vision/pull/7562#discussion_r1236702589